### PR TITLE
Get FishNet running

### PR DESCRIPTION
**Pull request by delventhalz**
With the changes in this PR, you should be able to run FishNet. This does not update `run_tests`, `run_docker_tests`, `run_lint`, or `run_bandit`. The deployment process will also be streamlined in future PRs.

To run, download the repo, and from the project directory:
```bash
docker-compose up
```

Then in a new process:
```bash
./bin/protogen
docker exec -it sawtooth-client-default bash
```

This will enter the sawtooth client container which has the `sawtooth_sdk` in the PYTHONPATH, which the transaction processor needs. So from within that container:
```bash
cd sawtooth-supply-chain/
./bin/transaction-processor
```

Open one more process:
```bash
cd client/
npm install
npm run build
cd ../server/
npm install
npm run init
npm run make-fish
curl localhost:8008/batches -X POST -H "Content-Type: application/octet-stream" --data-binary @fish.batch 
npm start
```

Now you should be able to navigate to `http://localhost:3000/fish` and see the FishNet client. Creating users, fish, and updating data should all work.

### CODE REVIEW

File: client/src/services/api.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ const m = require('mithril')
 const _ = require('lodash')
 const sjcl = require('sjcl')
 
-const STORAGE_KEY = 'tnt.authorization'
+const STORAGE_KEY = 'fishnet.authorization'
```
Comment (peterschwarz) [inline client/src/services/api.js]: This should be `supply_chain`

File: client/src/services/transactions.js [hunk 2]
```diff
@@ -23,11 +23,11 @@ const { signer, TransactionEncoder } = require('sawtooth-sdk/client')
 const modals = require('../components/modals')
 const api = require('../services/api')
 
-const STORAGE_KEY = 'tnt.encryptedKey'
+const STORAGE_KEY = 'fishnet.encryptedKey'
```
Comment (peterschwarz) [inline client/src/services/transactions.js]: ditto

File: docs/source/family_specification.rst [hunk 3]
```diff
@@ -404,7 +404,7 @@ first six characters of the SHA-512 hash of the string
    >>> get_hash('track_and_trade')[:6]
```
Comment (peterschwarz) [inline docs/source/family_specification.rst]: Should this get updated to `supply_chain`?  If not, perhaps this should be noted that it is `track_and_trade`, for historical reasons.

File: server/db/index.js [hunk 4]
```diff
@@ -22,7 +22,7 @@ const jsSchema = require('js-schema')
 
 const HOST = process.env.DB_HOST || 'localhost'
 const PORT = process.env.DB_PORT || 28015
-const NAME = process.env.DB_NAME || 'tnt'
+const NAME = process.env.DB_NAME || 'sc'
```
Comment (peterschwarz) [inline server/db/index.js]: This should be the full word for the db name: `supply_chain`, no reason this should be short - it's not informative enough.

File: server/scripts/bootstrap_database.js [hunk 5]
```diff
@@ -20,7 +20,7 @@ const r = require('rethinkdb')
 
 const HOST = process.env.DB_HOST || 'localhost'
 const PORT = process.env.DB_PORT || 28015
-const NAME = process.env.DB_NAME || 'tnt'
+const NAME = process.env.DB_NAME || 'sc'
```
Comment (peterschwarz) [inline server/scripts/bootstrap_database.js]: `supply_chain`

**Review by peterschwarz — CHANGES_REQUESTED** [review]

File: client/src/services/api.js (hunk 1 above)
Comment (delventhalz) [reply client/src/services/api.js]: It's part of the client, with runs on Sawtooth Supply Chain, but there could be than one client per Supply Chain server. So `fish_net.authorization`, `asset_track.authorization`, etc. I am feeling the underscores though . . .

Comment (peterschwarz) [reply client/src/services/api.js]: Ah, I see.  This is the storage key in `localStorage`.

**Review by peterschwarz — APPROVED** [review]

**Comment by vaporos** [comment]
We should use supply-chain-tp or sawtooth-supply-chain-tp as the script name, for a couple reasons:

a) will make more sense in "ps" output
b) approach is compatible with projects with more than one transaction processor, for which this might serve as an example

**Review by vaporos — APPROVED** [review]

**Comment by delventhalz** [comment]
Renamed to `supply_chain_tp`. Though I wonder now if the abbreviation "tp" is clear enough. Maybe "supply_chain_processor" would be more appropriate. In any case, merging it as is. Names can be nailed down later.

**Pull request merged**